integration/experiment/system_characterization/host_config.py

```diff
@@ -0,0 +1,93 @@
+#!/usr/bin/env python3
+#
+#  Copyright (c) 2015 - 2023, Intel Corporation
+#  SPDX-License-Identifier: BSD-3-Clause
+#
+
+import sys
+import json
+import glob
+import os
+from argparse import ArgumentParser
+
+
+def combine(directory, append_file=None):
+    """Combine a set of host specific parameter configuration files
+
+    Args:
+
+        directory (str): Directory containing host specific files,
+                         where each file is named after the hostname.
+                         Optionally files may contain the ".json"
+                         suffix.
+
+        append_file (str): Path to a configuration file that is not
+                           host specific which should also be combined.
+
+    """
+    if append_file:
+        with open(append_file) as fid:
+            output = json.load(fid)
+        if type(output) is not dict:
+            raise TypeError('Append file is not JSON object')
+    else:
+        output = dict()
+    host_files = glob.glob(f'{directory}/*')
+    for host_file in host_files:
+        host_name = os.path.basename(host_file).split('.json')[0]
+        try:
+            with open(host_file) as fid:
+                host_data = json.load(fid)
+        except json.decoder.JSONDecodeError:
+            sys.stderr.write(f'Warning: File {host_file} is not a JSON file, it will be ignored\n')
```

Comment:
Should we treat this (and other warnings below) as an error?

I'm thinking of a scenario where the user is running this within a script and some files get skipped (unintentionally) without the user noticing the warning message.

Reply:
I made these errors, but now we force the files to have the .json extension.  This will still allow for someone to populate the directory with a README or other documentation.